src/modules/cpu/kernel/normalize_generic.hpp

```diff
@@ -0,0 +1,513 @@
+
+            // Inv std dev calculations missing
+        if(axis_mask == 2)
+            normalize_2D_tensor_avx_axis2(srcPtrTemp, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, stdDevTensor, shift, srcAudioDims, paramStride);
+        else
+            normalize_2D_tensor(srcPtrTemp, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, stdDevTensor, shift, srcAudioDims, paramStride);
+        }
+        else if(nDim == 3)
+        {
+            if(axis_mask == 1) // Normalize across Channels
+            {
+                paramStride[0] = paramStride[1] = 1;
+                paramStride[2] = 0;
+            }
+            else if(axis_mask == 2) // Normalize across Width
+            {
+                //TODO
+            }
+            else if(axis_mask == 3) // Normalize across Height
+            {
+                //TODO
+            }
+            srcPtrChannel = srcPtrTemp + (begin[0] * layoutParams.bufferMultiplier);
+            for(int i = 1; i < nDim; i++)
+                srcPtrChannel += begin[i] * srcGenericDescPtr->strides[i];
+
+            if((axis_mask == 1) && (srcGenericDescPtr->layout == RpptLayout::NHWC) && (dstGenericDescPtr->layout == RpptLayout::NHWC) && (srcGenericDescPtr->dims[3] == 16))
+                normalize_3D_tensor_avx_axis1(srcPtrChannel, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, stdDevTensor, scale, shift, paramStride, length[1] * layoutParams.bufferMultiplier);
+            else if((axis_mask == 1) && (srcGenericDescPtr->layout == RpptLayout::NHWC) && (dstGenericDescPtr->layout == RpptLayout::NHWC))
+                normalize_3D_tensor_axis1_nontoggle(srcPtrChannel, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, stdDevTensor, scale, shift, paramStride);
+            else if((axis_mask == 1) && (srcGenericDescPtr->layout == RpptLayout::NHWC) && (dstGenericDescPtr->layout == RpptLayout::NCHW))
+                normalize_3D_tensor_axis1_toggle(srcPtrChannel, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, stdDevTensor, scale, shift, paramStride);
+        }
+        else
+        {
+            srcPtrChannel = srcPtrTemp + (begin[0] * layoutParams.bufferMultiplier);
+            if(axis_mask == 1) // Normalize across Channels
+            {
+                for(int i = 0; i < nDim; i++)
+                {
+                    paramStride[i] = (i == nDim - 1)? 0 : 1;
+                    srcPtrChannel += begin[i + 1] * srcGenericDescPtr->strides[i + 1];
+                }
+            }
+
+            Rpp32f multiplier[srcGenericDescPtr->dims[nDim]];
+            for(int i = 0; i < srcGenericDescPtr->dims[nDim]; i++)
+                multiplier[i] = scale / stdDevTensor[i];
+            if(axis_mask == 1)
+                normalize_ND_tensor_axis1_nontoggle(srcPtrChannel, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, multiplier, shift, paramStride, length, nDim, 1);
```

Comment:
For try to add a generic kernel that can normalize for any axis_mask value

Reply:
Yes, thats in progress.